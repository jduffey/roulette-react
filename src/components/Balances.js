import { ethers } from "ethers";
import { useState } from "react";
import { useEffect } from "react";

const provider = new ethers.providers.JsonRpcProvider("http://localhost:8545");

const TOKEN_CONTRACT_ADDRESS = "0x73511669fd4dE447feD18BB79bAFeAC93aB7F31f";

async function getEthBalance(provider, address) {
    const signer = provider.getSigner(address);
    const balance = await signer.getBalance();
    return ethers.utils.formatEther(balance);
}

// TODO leaving this for reference
// async function sendEther(provider, from, to, amount) {
//     const signer = provider.getSigner(from);
//     const tx = await signer.sendTransaction({
//         to: to,
//         value: ethers.utils.parseEther(amount)
//     });
//     return tx;
// }

async function getBlock(provider) {
    const block = await provider.getBlock();
    return block;
}

async function getTokenBalance(provider, address, tokenAddress) {
    const signer = provider.getSigner(address);
    const token = new ethers.Contract(tokenAddress, [
        "function balanceOf(address) view returns (uint)",
    ], signer);
    const balance = await token.balanceOf(address);
    const formattedBalance = ethers.utils.formatEther(balance);
    return formattedBalance;
}

async function depositEther(provider, from, tokenAddress, amount) {
    const signer = provider.getSigner(from);
    const token = new ethers.Contract(tokenAddress, [
        "function deposit() payable",
    ], signer);
    const tx = await token.deposit({
        value: ethers.utils.parseEther(amount)
    });
    return tx;
}

export function Balances() {
    const [ethBalances, setEthBalances] = useState([]);
    const [tokenBalances, setTokenBalances] = useState([]);

    const [block, setBlock] = useState({});

    const [rerender, setRerender] = useState(false);

    const [tokenSymbol, setTokenSymbol] = useState("");

    useEffect(() => {
        // Default values, including seed phrase: https://hardhat.org/hardhat-network/docs/reference
        const addresses = [
            "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
            "0x70997970C51812dc3A010C7d01b50e0d17dc79C8",
            "0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC",
            "0x90F79bf6EB2c4f870365E785982E1f101E93b906",
            "0x15d34AAf54267DB7D7c367839AAf71A00a2C6A65",
            "0x9965507D1a55bcC2695C58ba16FB37d819B0A4dc",
            "0x976EA74026E726554dB657fA54763abd0C3a0aa9",
            "0x14dC79964da2C08b23698B3D3cc7Ca32193d9955",
            "0x23618e81E3f5cdF7f54C3d65f7FBc0aBf5B21E8f",
            "0xa0Ee7A142d267C1f36714E4a8F75612F20a79720",
            "0xBcd4042DE499D14e55001CcbB24a551F3b954096",
            "0x71bE63f3384f5fb98995898A86B02Fb2426c5788",
            "0xFABB0ac9d68B0B445fB7357272Ff202C5651694a",
            "0x1CBd3b2770909D4e10f157cABC84C7264073C9Ec",
            "0xdF3e18d64BC6A983f673Ab319CCaE4f1a57C7097",
            "0xcd3B766CCDd6AE721141F452C550Ca635964ce71",
            "0x2546BcD3c84621e976D8185a91A922aE77ECEc30",
            "0xbDA5747bFD65F08deb54cb465eB87D40e51B197E",
            "0xdD2FD4581271e230360230F9337D5c0430Bf44C0",
            "0x8626f6940E2eb28930eFb4CeF49B2d1F2C9C1199",
        ];

        const addressWithBalancePromises = addresses.map(async (address) => {
            const balance = await getEthBalance(provider, address);
            return { address, balance };
        });

        Promise.all(addressWithBalancePromises).then((res) => {
            const newBalances = res.reduce((acc, cur) => {
                acc[cur.address] = cur.balance;
                return acc;
            }, {});
            setEthBalances(newBalances);
        });

        // TODO figure out how to get the token symbol just once.. no need to do this every time we render obviously
        const token = new ethers.Contract(
            TOKEN_CONTRACT_ADDRESS,
            ["function symbol() view returns (string)"],
            provider);
        token.symbol().then((symbol) => {
            setTokenSymbol(symbol);
        });

        const addressesWithTokenBalancePromises = addresses.map(async (address) => {
            const balance = await getTokenBalance(provider, address, TOKEN_CONTRACT_ADDRESS);
            return { address, balance };
        });

        Promise.all(addressesWithTokenBalancePromises).then((res) => {
            const newBalances = res.reduce((acc, cur) => {
                acc[cur.address] = cur.balance;
                return acc;
            }, {});
            setTokenBalances(newBalances);
        });

        getBlock(provider)
            .then((block) => {
                setBlock(block);
            });

    }, [rerender]);

    const combinedBalances = Object.keys(ethBalances).reduce((acc, address) => {
        acc[address] = {
            ethBalance: ethBalances[address],
            tokenBalance: tokenBalances[address]
        };
        return acc;
    }, {});

    return (
        <div
            style={{
                color: "white",
                fontFamily: "monospace",
                fontSize: "1.25rem",
            }}
        >
            <div>
                {
                    ["hash", "number", "timestamp"].map((key) => {
                        return (
                            <div key={key}>
                                {`${key}: ${block[key]}`}
                            </div>
                        );
                    })
                }
            </div>
            <div>
                <table>
                    <thead>
                        <tr>
                            <th>Address</th>
                            <th>ETH Balance</th>
                            <th>{tokenSymbol} Balance</th>
                            <th>Deposit 1 ETH</th>
                        </tr>
                    </thead>
                    <tbody>
                        {
                            Object.entries(combinedBalances).map(([addr, { ethBalance, tokenBalance }]) => {
                                return (
                                    <tr key={addr}>
                                        <td>{`${addr.slice(0, 6)}..${addr.slice(-4)}`}</td>
                                        <td>{Number(ethBalance).toFixed(18).padStart(18 + 6, String.fromCharCode(160))}</td>
                                        <td>{Number(tokenBalance).toFixed(18).padStart(18 + 10, String.fromCharCode(160))}</td>
                                        <td>
                                            <button
                                                onClick={() => {
                                                    depositEther(
                                                        provider,
                                                        addr,
                                                        TOKEN_CONTRACT_ADDRESS,
                                                        "1"
                                                    ).then(() => {
                                                        setRerender(!rerender);
                                                    });
                                                }}
                                            >
                                                Dep 1 ETH
                                            </button>
                                        </td>
                                    </tr>
                                );
                            })
                        }
                    </tbody>
                </table>

            </div>
        </div>
    )
}
