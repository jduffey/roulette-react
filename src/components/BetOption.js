import { Chip } from "./Chip";
import { BETTING_SQUARE_OUTLINE_SIZE } from "../common/project-globals";

export function BetOption(props) {
    return (
        <div
            className={`betting-square-visible betting-square-${props.betName}`}
            onClick={props.onClick}
            style={{
                left: props.styleData.left,
                top: props.styleData.top,
                height: props.styleData.height - BETTING_SQUARE_OUTLINE_SIZE,
                width: props.styleData.width - BETTING_SQUARE_OUTLINE_SIZE,
                backgroundColor: props.styleData.backgroundColor,
            }}
        >
            <div className="betting-square-contents">
                <div
                    className={`betting-square-label ${props.classNamePrefix}-square-label`}
                    style={{
                        backgroundColor: props.styleData.labelBackgroundColor,
                    }}
                >
                    {props.displayLabel}
                </div>
                <Chip
                    key={props.chipAmount}
                    onClick={props.onClick}
                    className="betting-square-chip"
                    chipAmount={props.betAmount}
                />
            </div>
        </div>
    );
}
