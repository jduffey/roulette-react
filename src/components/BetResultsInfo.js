import { WINNING_CRITERIA } from "../common/winning-criteria";

const className = "bet-results-info";
export function BetResultsInfo(props) {
    const sumBetAmounts = Object.keys(props.bets).reduce((acc, betOption) => {
        const betAmountOnBet = props.bets[betOption];
        return acc + betAmountOnBet;
    }, 0);

    const sumWinnings = Object.keys(props.bets).reduce((acc, betOption) => {
        const winningsOnBet = calculateWinningsOnBet(props, betOption);
        return acc + winningsOnBet;
    }, 0);

    const sumBetsReturned = Object.keys(props.bets).reduce((acc, betOption) => {
        const betReturnedAmount = calculateBetReturnedAmount(props, betOption);
        return acc + betReturnedAmount;
    }, 0);

    const startingBalanceText = props.startingBalance ?
        "$ " + props.startingBalance.toString() :
        "";

    const netChangeInBalance = sumWinnings + sumBetsReturned - sumBetAmounts;
    const netChangeInBalanceText = props.startingBalance ?
        "$ " + netChangeInBalance.toString() :
        "";

    const finalBalance = props.startingBalance + netChangeInBalance;
    const finalBalanceText = props.startingBalance ?
        "$ " + finalBalance.toString() :
        "";

    return (
        <div className={className}>
            <div
                className="bet-info-table-title"
            >
                PREVIOUS ROUND RESULTS
            </div>
            <table className="bet-results-info-table">
                <tbody>
                    <tr style={{
                        textAlign: "center",
                    }}>
                        <th>Bet Name</th>
                        <th>Bet Amount</th>
                        <th>Winnings</th>
                        <th>Bet Returned</th>
                    </tr>
                    {Object.keys(props.bets).map((betOption) => {
                        const betAmountOnBet = props.bets[betOption];

                        let winningsOnBet = calculateWinningsOnBet(props, betOption);
                        const winningsOnBetText = winningsOnBet ? "$ " + winningsOnBet.toString() : "";

                        let betReturnedAmount = calculateBetReturnedAmount(props, betOption);
                        const betReturnedAmountText = betReturnedAmount ? "$ " + betReturnedAmount.toString() : "";

                        return (
                            <tr key={betOption}>
                                <td>{betOption}</td>
                                <td className="bet-results-info-table-bet-amount">{"$ " + betAmountOnBet.toString()}</td>
                                <td>{winningsOnBetText}</td>
                                <td>{betReturnedAmountText}</td>
                            </tr>
                        );
                    })}
                    <tr style={{ height: "10px" }}></tr>
                    <tr>
                        <td>TOTALS</td>
                        <td>{"$ " + sumBetAmounts}</td>
                        <td>{"$ " + sumWinnings}</td>
                        <td>{"$ " + sumBetsReturned}</td>
                    </tr>
                    <tr style={{ height: "10px" }}></tr>
                    <tr className="balance-change-value">
                        <td>
                            Starting
                        </td>
                        <td colSpan={3}>
                            {startingBalanceText}
                        </td>
                    </tr>
                    <tr className="balance-change-value">
                        <td>
                            Net
                        </td>
                        <td colSpan={3}>
                            {netChangeInBalanceText}
                        </td>
                    </tr>
                    <tr className="balance-change-value">
                        <td>
                            Final
                        </td>
                        <td colSpan={3}>
                            {finalBalanceText}
                        </td>
                    </tr>
                </tbody>
            </table>
        </div>
    );
}

function calculateWinningsOnBet(props, betOption) {
    const multipliers = {
        "1": 35,
        "2": 35,
        "3": 35,
        "4": 35,
        "5": 35,
        "6": 35,
        "7": 35,
        "8": 35,
        "9": 35,
        "10": 35,
        "11": 35,
        "12": 35,
        "13": 35,
        "14": 35,
        "15": 35,
        "16": 35,
        "17": 35,
        "18": 35,
        "19": 35,
        "20": 35,
        "21": 35,
        "22": 35,
        "23": 35,
        "24": 35,
        "25": 35,
        "26": 35,
        "27": 35,
        "28": 35,
        "29": 35,
        "30": 35,
        "31": 35,
        "32": 35,
        "33": 35,
        "34": 35,
        "35": 35,
        "36": 35,
        "0": 35,
        "00": 35,
        "1st 12": 2,
        "2nd 12": 2,
        "3rd 12": 2,
        "Top Row": 2,
        "Middle Row": 2,
        "Bottom Row": 2,
        "1 to 18": 1,
        "19 to 36": 1,
        "Even": 1,
        "Odd": 1,
        "Red": 1,
        "Black": 1,
    };

    const betAmount = props.bets[betOption];

    return WINNING_CRITERIA[props.winningWheelNumber].includes(betOption) ?
        betAmount * multipliers[betOption] :
        0;
}

function calculateBetReturnedAmount(props, betOption) {
    return WINNING_CRITERIA[props.winningWheelNumber].includes(betOption) ?
        props.bets[betOption] :
        0;
}
