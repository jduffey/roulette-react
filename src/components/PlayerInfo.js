import React from 'react';

const CLASS_NAME = "PlayerInfo-component";
export function PlayerInfo(props) {
    return (
        <div
            className={CLASS_NAME}
        >
            <div>
                {"Avail. Balance"}
                < br />
                {`$ ${props.availableBalance.toLocaleString()}`}
            </div>
            <div>
                {"Total Bet"}
                < br />
                {`$ ${props.totalBetAmount.toLocaleString()}`}
            </div>
            <div
                id="reset-history"
                onClick={() => props.onClick()}
            >
                RESET HISTORY
            </div>
        </div >
    )
}
