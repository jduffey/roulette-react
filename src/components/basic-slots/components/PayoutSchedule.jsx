export function PayoutSchedule() {
    return (
        <div
            style={{
                width: 516,
                height: 444,
                borderRadius: 30,
                backgroundColor: '#00B5FF',
            }}
        >
            <div
                style={{
                    width: 516,
                    height: 60,
                    display: 'flex',
                    justifyContent: 'center',
                    alignItems: 'center',
                    fontSize: 40,
                    fontWeight: 'bold',
                    borderTopLeftRadius: 30,
                    borderTopRightRadius: 30,
                    rotate: '-4deg',
                    color: '#FFE24B',
                    '-webkit-text-stroke-color': 'black',
                    '-webkit-text-stroke-width': '2px',
                }}
            >
                <StarSvg /> WINNING <StarSvg />
            </div>
            <div
                style={{
                    width: 516,
                    height: 60,
                    display: 'flex',
                    justifyContent: 'center',
                    alignItems: 'center',
                    fontSize: 24,
                    fontWeight: 'bold',
                    color: 'white',
                    '-webkit-text-stroke-color': 'black',
                    '-webkit-text-stroke-width': '2px',
                }}
            >
                COMBINATIONS
            </div>
            <div
                style={{
                    padding: 2,
                }}
            >
                <SlotIconSeven />
                <SlotIconSeven />
                <SlotIconSeven />
            </div>
            <div
                style={{
                    width: 516,
                    height: 160,
                    display: 'flex',
                    justifyContent: 'center',
                    alignItems: 'center',
                    fontSize: 24,
                    fontWeight: 'bold',
                    backgroundColor: 'white',
                }}
            >
                <div
                    style={{
                        width: 516,
                        display: 'flex',
                        justifyContent: 'right',
                        alignItems: 'center',
                        fontSize: 70,
                        fontWeight: 'bold',
                        color: '#74EE15',
                        rotate: '6deg',
                    }}
                >
                    $2000
                </div>
            </div>
        </div >
    );
}

const StarSvg = () => (
    <svg width="48" height="46" viewBox="0 0 48 46" fill="none" xmlns="http://www.w3.org/2000/svg">
        <path d="M24 4.8541L28.1863 17.7381L28.523 18.7746H29.6129H43.1599L32.2001 26.7373L31.3184 27.3779L31.6552 28.4144L35.8415 41.2984L24.8817 33.3356L24 32.695L23.1183 33.3356L12.1585 41.2984L16.3448 28.4144L16.6816 27.3779L15.7999 26.7373L4.84011 18.7746H18.3871H19.477L19.8137 17.7381L24 4.8541Z" fill="#FFE24B" stroke="black" stroke-width="3" />
    </svg>
);

const SlotIconSeven = () => (
    <svg width="70" height="72" viewBox="0 0 70 72" fill="none" xmlns="http://www.w3.org/2000/svg">
        <path d="M16.4859 6.26481V4.76481H14.9859H3H1.5V6.26481V30.2718V31.7718H3H14.9859H16.3751L16.4815 30.3867L16.6935 27.6263L17.4399 25.3837L19.3003 23.3342L21.4464 22.2596H24.9915L27.618 22.8666L29.9686 23.9366L29.9784 23.9411L29.9881 23.9453L33.6065 25.5307L33.6568 25.5527L33.7086 25.5711L37.5531 26.9299L37.7157 26.9874L37.8871 27.0065L43.9931 27.6859L44.1313 27.7013L44.27 27.691L46.8968 27.4961L37.5835 34.7504L37.56 34.7688L37.5372 34.7881L30.3004 40.903L30.2364 40.9571L30.1789 41.018L26.1082 45.3211L25.9988 45.4368L25.9161 45.5728L22.75 50.7819L22.6766 50.9026L22.6271 51.035L20.5917 56.4705L20.5473 56.5892L20.5234 56.7136L19.3926 62.6021L19.3719 62.7101L19.3671 62.82L19.141 68.029L19.1242 68.4168L19.2975 68.7641L19.7498 69.67L20.1642 70.5H21.0919H47.0989H47.4072L47.6906 70.3784L49.2736 69.699L50.182 69.3091V68.3206V59.7143V55.5229L50.8191 51.2689L52.3033 47.4468L54.2559 43.9705L58.493 37.7171L63.6681 30.9571L63.8051 30.7782L63.8834 30.5669L66.8233 22.6401L66.8674 22.5213L66.8909 22.3969L68.4739 14.0171L68.5 13.8791V13.7387V6.26481V4.76481H67H55.4664H54.1529L53.9795 6.06685L53.5543 9.26059L52.6676 11.3328H52.4572L50.3574 10.1857L45.2054 6.82079L45.1454 6.78161L45.082 6.74833L40.3329 4.25704L40.24 4.20831L40.1412 4.17299L36.9751 3.04058L36.8257 2.98716L36.6685 2.96616L33.2763 2.5132L33.0839 2.48752L32.8914 2.51162L29.273 2.96458L29.0912 2.98734L28.9202 3.05321L25.9803 4.18561L25.8896 4.22054L25.8042 4.26687L20.3766 7.21112L20.3512 7.22491L20.3263 7.23967L16.4859 9.51885V6.26481Z" fill="#BB2828" stroke="black" stroke-width="3" />
    </svg>
);