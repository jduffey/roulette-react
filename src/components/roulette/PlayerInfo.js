import { useEffect, useState } from "react";

import {
    getTokenBalance,
    FIRST_PLAYER_ADDRESS,
} from '../../common/blockchainWrapper';

import { ethers } from "ethers";

const wsProvider = new ethers.providers.WebSocketProvider("ws://localhost:8545");

console.log("wsProvider:", wsProvider);

const CLASS_NAME = "PlayerInfo-component";
export function PlayerInfo(props) {
    const [tokenBalance, setTokenBalance] = useState("Not loaded...");

    // Adapted from https://github.com/ethers-io/ethers.js/issues/1053#issuecomment-808736570
    const EXPECTED_PONG_BACK = 2000
    const KEEP_ALIVE_CHECK_INTERVAL = 1000

    let startCounter = 1;
    const startConnection = () => {
        if (typeof wsProvider._websocket === "undefined") {
            console.log("Websocket connection is undefined. Is the blockchain node running?");
            return;
        }

        // TODO why does this counter always make it to 2 (twice)?
        console.log(`Websocket connection started ${startCounter++} time(s).`);

        wsProvider._websocket.onmessage = (event) => {
            console.log("An event was received:", event);
        }

        let pingTimeout = null
        let keepAliveInterval = null
        wsProvider._websocket.onopen = () => {
            console.log("onopen");
            keepAliveInterval = setInterval(() => {
                console.log("Running the interval", Date.now());
                (async (addr) => {
                    console.log("Getting token balance", Date.now());
                    const balance = await getTokenBalance(addr);
                    const balanceAsNumber = Number(balance);
                    setTokenBalance(balanceAsNumber);
                })(FIRST_PLAYER_ADDRESS);
                pingTimeout = setTimeout(() => {
                    console.log("setting what happens if the ping times out", Date.now());
                    wsProvider._websocket.close();
                }, EXPECTED_PONG_BACK)
            }, KEEP_ALIVE_CHECK_INTERVAL)
        }

        wsProvider._websocket.onclose = () => {
            console.log("onclose");
            clearInterval(keepAliveInterval);
            clearTimeout(pingTimeout);
            startConnection();
        }
    }

    startConnection();

    return (
        <div
            className={CLASS_NAME}
        >
            <div>
                {"Tok. Balance"}
                < br />
                {tokenBalance.toLocaleString()}
            </div>
            <div>
                {"Avail. Balance"}
                < br />
                {`$ ${props.availableBalance.toLocaleString()}`}
            </div>
            <div>
                {"Total Bet"}
                < br />
                {`$ ${props.totalBetAmount.toLocaleString()}`}
            </div>
            <div
                id="reset-history"
                onClick={() => props.onClick()}
            >
                RESET HISTORY
            </div>
        </div >
    )
}
