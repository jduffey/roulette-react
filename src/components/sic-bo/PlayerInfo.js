import React from 'react';

const CLASS_NAME = "SicBo-PlayerInfo-component";
export function PlayerInfo() {
    const ownedBalance = 10000;
    const chipsInHand = ownedBalance;
    const totalBetsOnBoard = 0;

    return (
        <div
            className={CLASS_NAME}
        >
            <div>
                {"Owned Bal."}
                < br />
                {ownedBalance}
            </div>
            <div>
                {"Chips in Hand"}
                < br />
                {chipsInHand}
            </div>
            <div>
                {"Total Bet"}
                < br />
                {totalBetsOnBoard}
            </div>
        </div >
    )
}
