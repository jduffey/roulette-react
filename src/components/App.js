import { Route, Routes } from "react-router-dom";

import { Layout } from "./Layout";

import { Home } from "./Home";
import { Roulette } from "./roulette/Roulette";
import { SicBo } from "./SicBo";
import { Balances } from "./balances/Balances";
import { EventsDashboard } from "./EventsDashboard";

import {
    FIRST_PLAYER_ADDRESS,
    SECOND_PLAYER_ADDRESS,
    THIRD_PLAYER_ADDRESS,
} from "../common/blockchainWrapper";

const elementInLayout = (element) => <Layout>{element}</Layout>

export function App() {
    return (
        <Routes>
            <Route path="/" element={elementInLayout(<Home />)} />
            <Route path="/roulette-p1" element={elementInLayout(<Roulette playerAddress={FIRST_PLAYER_ADDRESS} playerDbEndpoint="/player-1" />)} />
            <Route path="/roulette-p2" element={elementInLayout(<Roulette playerAddress={SECOND_PLAYER_ADDRESS} playerDbEndpoint="/player-2" />)} />
            <Route path="/roulette-p3" element={elementInLayout(<Roulette playerAddress={THIRD_PLAYER_ADDRESS} playerDbEndpoint="/player-3" />)} />
            <Route path="/sic-bo" element={elementInLayout(<SicBo />)} />
            <Route path="/balances" element={elementInLayout(<Balances />)} />
            <Route path="/events" element={elementInLayout(<EventsDashboard />)} />
        </Routes>
    );
}
