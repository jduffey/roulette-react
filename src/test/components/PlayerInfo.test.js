import renderer from 'react-test-renderer';

import { PlayerInfo } from '../../components/PlayerInfo';

describe('PlayerInfo', () => {
    it.each([
        ["foo", "bar"], // TODO shouldn't pass non-numeric values but still renders for now
        [123456, 56789], // TODO this test will fail if run in an environment with a different locale (e.g. using a period for thousands separator)
    ])('renders when availableBalance is %s and totalBetAmount is %s', (availableBalance, totalBetAmount) => {
        const sut =
            <PlayerInfo
                availableBalance={availableBalance}
                totalBetAmount={totalBetAmount}
            />;

        const actual = renderer.create(sut);

        expect(actual).toMatchSnapshot();
    });
});
