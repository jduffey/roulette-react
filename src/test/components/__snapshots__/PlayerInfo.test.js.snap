// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerInfo renders when availableBalance is 123456 and totalBetAmount is 56789 1`] = `
<div
  className="PlayerInfo-component"
>
  <div>
    Avail. Balance
    <br />
    $ 123,456
  </div>
  <div>
    Total Bet
    <br />
    $ 56,789
  </div>
  <div
    id="reset-history"
    onClick={[Function]}
  >
    RESET HISTORY
  </div>
</div>
`;

exports[`PlayerInfo renders when availableBalance is foo and totalBetAmount is bar 1`] = `
<div
  className="PlayerInfo-component"
>
  <div>
    Avail. Balance
    <br />
    $ foo
  </div>
  <div>
    Total Bet
    <br />
    $ bar
  </div>
  <div
    id="reset-history"
    onClick={[Function]}
  >
    RESET HISTORY
  </div>
</div>
`;
